ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -335,6 +340,8 @@ private PutChunk getChunkToFill() {
         chunkCounter++;
         chunk.prepareForBuilding(chunkCounter, getSizeOfChunkAt(chunkCounter));
         chunkToFill = chunk;
+        routerMetrics.chunkFillerWaitingTrunkTimeMs.update(time.milliseconds() - timeFilledPreviousChunkMs);
```

Comment:
I don't think this is correct. You are updating `timeFilledPreviousChunkMs` when the fill is complete and use that as the threshold for how much we wait. But once the fill is complete, we can't reuse the PutChunk until the chunk it holds is sent out, and we seem to end up calculating that in the "wait time"